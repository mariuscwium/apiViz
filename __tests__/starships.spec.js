import React from 'react';
import Starships from '../app/components/explorer/starships.jsx';
import renderer from 'react-test-renderer';

describe('Testing Starships component', () => {
    beforeEach(() => {
        fetch.resetMocks();
    });

    test('Displays when mounted', async () => {
        fetch.mockResponseOnce(JSON.stringify({ name: 'Imperial shuttle' }));

        const selectedCharacter = {
            starships: [test]
        };

        let promise;
        const addPromise = newPromise => {
            promise = newPromise;
        };

        const component = renderer.create(
            <Starships selectedCharacter={selectedCharacter} addPromise={addPromise}>
                {state => {
                    if (state.data) {
                        return <div>{state.data.name}</div>;
                    } else {
                        return null;
                    }
                }}
            </Starships>
        );
        await promise;
        const instance = component.getInstance();
        instance.setState({ loading: false });

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
