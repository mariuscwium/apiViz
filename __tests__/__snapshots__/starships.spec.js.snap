// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Starships component Displays when mounted 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Character starships
  </h1>
  <ul
    className="sc-bdVaJa jJjsEG"
  >
    <li
      className="sc-bwzfXH fQvBvU"
    >
      Imperial shuttle
    </li>
  </ul>
</div>
`;
